cadence/contracts/bridge/FlowEVMBridgeConfig.cdc

```diff
@@ -29,13 +30,15 @@ contract FlowEVMBridgeConfig {
     /// Default ERC20.decimals() value
     access(all)
     let defaultDecimals: UInt8
+    access(all)
+    var gasLimit: UInt64
```

Comment:
I see that we're using this everywhere now instead of before having different gas limits for different types of operations. Correct me if I'm wrong, but I thought that in EVM, the gas limit is completely consumed regardless of whether or not the transaction actually needed it or not, so would it make more sense to have different gas limits for different types of transactions? That way, we could make sure they are all only using as much as is needed

Reply:
Hmm good callout, let me do some digging here. So far I've used the [gas limit](https://ethereum.org/en/developers/docs/gas/#what-is-gas-limit) value as the maximum computation allowable for a given operation, which while the final computation used is a factor in fee calculation is not necessarily the basis for fees which have their own rules in relation to operations in Cadence. So from my assumptions, I've considered the `gasLimit` is the upper bound for how much computation is allowable for a COA call. If it's true that the `gasLimit` is in fact defining a set fee amount for the execution of the call or even the gas that is exactly to be used by that call, then you're right that should be updated for more granular control of user fees but I need to double check my understanding.

Reply:
From running a quick EVM deployment transaction in emulator with this gas limit, I can see that the gas limit seems to be an upper bound as the gas consumed is much less than the provided gas limit value. In this case, I don't necessarily see the need for a more granular gas limit value, especially since the fees are ultimately all paid via the wrapping flow transaction - callers could always add a gas limit to the Cadence transaction which would limit the total gas used for the entire transaction. What do you think?